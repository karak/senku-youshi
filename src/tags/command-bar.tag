<command-bar>
  <button ref="shuffleButton" onclick={ shuffleButtonClick }>並べ替え</button>
  <script>
  shuffleButtonClick() {
    this.trigger('shuffle');
  }
  </script>
</command-bar>
