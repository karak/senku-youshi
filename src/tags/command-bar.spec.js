import './command-bar.tag';
import { shallow } from 'riot-test-utils';
import { simulate } from 'simulate-event';

describe('<command-bar />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow('command-bar');
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it('has "shuffleButton" in its refs', () => {
    expect(wrapper.instance().refs).toHaveProperty('shuffleButton');
  });

  it('shuffleButton has a certain label', () => {
    const shuffleButton = wrapper.instance().refs.shuffleButton;

    expect(shuffleButton.textContent).toBe('並べ替え');
  });

  it('fires "shuffle" event when button is clicked', () => {
    const handler = jest.fn();

    const shuffleButton = wrapper.instance().refs.shuffleButton;

    wrapper.instance().on('shuffle', handler);

    simulate(shuffleButton, 'click');

    expect(handler).toHaveBeenCalled();
  });
});
