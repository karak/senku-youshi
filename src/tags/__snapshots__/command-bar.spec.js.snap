// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<command-bar /> matches snapshot 1`] = `
<command-bar
  data-is={
    Array [
      "command-bar",
    ]
  }
>
  <nav
    aria-label={
      Array [
        "main navigation",
      ]
    }
    class={
      Array [
        "navbar",
      ]
    }
    role={
      Array [
        "navigation",
      ]
    }
  >
     
    <div
      class={
        Array [
          "navbar-brand",
        ]
      }
    >
       
      <div
        class={
          Array [
            "navbar-item",
          ]
        }
      >
         
        <button
          class={
            Array [
              "button is-outlined",
            ]
          }
          ref={
            Array [
              "shuffleButton",
            ]
          }
        >
          並べ替え
        </button>
         
      </div>
       
      <div
        class={
          Array [
            "navbar-item",
          ]
        }
      >
         
        <button
          class={
            Array [
              "button is-primary",
            ]
          }
          ref={
            Array [
              "copyButton",
            ]
          }
        >
          クリップボードにコピー
        </button>
         
      </div>
       
    </div>
     
    <div
      class={
        Array [
          "navbar-menu",
        ]
      }
    >
       
    </div>
     
  </nav>
</command-bar>
`;
